Mathlib/Analysis/NormedSpace/Star/ContinuousFunctionalCalculus/Restrict.lean

```diff
@@ -0,0 +1,117 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import Mathlib.Topology.Algebra.Algebra
+import Mathlib.Topology.ContinuousFunction.Compact
+import Mathlib.Topology.ContinuousFunction.FunctionalCalculus
+
+/-! # Restriction of the continuous functional calculus to a scalar subring
+
+The main declaration in this file is:
+
++ `SpectrumRestricts.cfc`: builds a continuous functional calculus over a subring of scalars.
+  This is use for automatically deriving the continuous functional calculi on selfadjoint or
+  positive elements from the one for normal elements.
+
+This will allow us to take an instance of the
+`ContinuousFunctionalCalculus ℂ IsStarNormal` and produce both of the instances
+`ContinuousFunctionalCalculus ℝ (IsSelfAdjoint` and `ContinuousFunctionalCalculus ℝ≥0 (0 ≤ ·)`
+simply by proving:
+
+1. `IsSelfAdjoint x ↔ IsStarNormal x ∧ SpectrumRestricts Complex.re x`,
+2. `0 ≤ x ↔ IsSelfAdjoint x ∧ SpectrumRestricts Real.toNNReal x`.
+-/
+
+namespace SpectrumRestricts
+
+lemma compactSpace {R S A : Type*} [CommSemiring R] [CommSemiring S] [Ring A]
+    [Algebra R S] [Algebra R A] [Algebra S A] [IsScalarTower R S A] [TopologicalSpace R]
+    [TopologicalSpace S] {a : A} (f : C(S, R)) (h : SpectrumRestricts a f)
+    [h_cpct : CompactSpace (spectrum S a)] : CompactSpace (spectrum R a) := by
```

Comment:
Could this lemma and/or `starAlgHom` below be moved to the file with the rest of the `SpectrumRestricts` material, or would it screw up the imports?

Reply:
Yes, it screws up imports because `SpectrumRestricts` is in `Algebra.Algebra.Spectrum`, and has no topology as a dependency.